google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/RpcMethod.java

```diff
@@ -0,0 +1,167 @@
+    }
+
+    enum bucket_acl implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list,
+      patch,
+      update;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum buckets implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list,
+      patch,
+      update,
+      getIamPolicy,
+      lockRetentionPolicy,
+      setIamPolicy,
+      testIamPermissions;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum default_object_acl implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list,
+      patch,
+      update;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum hmacKey implements RpcMethod {
+      delete,
+      get,
+      list,
+      update,
+      create;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum notifications implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum object_acl implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list,
+      patch,
+      update;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum objects implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list,
+      patch,
+      update,
+      compose,
+      rewrite,
+      copy;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum serviceaccount implements RpcMethod {
+      get,
+      put;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    // create a map, which can be used to do a reverse lookup of an RpcMethod by its associated
+    // string value.
+    static final Map<String, RpcMethod> lookup =
+        Stream.<Stream<RpcMethod>>of(
+                Arrays.stream(bucket_acl.values()),
+                Arrays.stream(buckets.values()),
+                Arrays.stream(default_object_acl.values()),
+                Arrays.stream(hmacKey.values()),
+                Arrays.stream(notifications.values()),
+                Arrays.stream(object_acl.values()),
+                Arrays.stream(objects.values()),
+                Arrays.stream(serviceaccount.values()))
+            .flatMap(Function.identity()) // .flatten()
+            .collect(Collectors.toMap(RpcMethod::getFullyQualifiedMethodName, Function.identity()));
```

Comment:
Wait, why is there a double colon? Is this C++ in Java or did I miss this Java syntax in a newer Java version?

Reply:
Double colons were introduced in Java 8!